Urban-Fun/View Controllers/ProfileViewController.m

```diff
@@ -16,20 +17,35 @@ @interface ProfileViewController () <UICollectionViewDataSource, UICollectionVie
 @property (strong, nonatomic) IBOutlet UILabel *followerCount;
 @property (strong, nonatomic) IBOutlet UILabel *followingCount;
 @property (strong, nonatomic) IBOutlet UIButton *followButton;
+@property (strong, nonatomic) IBOutlet UICollectionView *collectionView;
+@property (strong, nonnull) NSArray *activitiesByUser;
 @end
 
 @implementation ProfileViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    self.collectionView.dataSource = self;
+    self.collectionView.delegate = self;
     // Do any additional setup after loading the view.
+    UIRefreshControl *refreshControl = [[UIRefreshControl alloc] init];
+    [refreshControl addTarget:self action:@selector(beginRefresh:) forControlEvents:UIControlEventValueChanged];
+    [self.collectionView insertSubview:refreshControl atIndex:0];
     self.currentUser = [User currentUser];
     if (self.profileToView == nil){
         self.profileToView = [User currentUser];
     }
     [self setUpView];
+    [self getUserActivities];
+    [self.collectionView reloadData];
 }
 
+- (void)beginRefresh:(UIRefreshControl *)refreshControl {
+    [self getUserActivities];
+    [refreshControl endRefreshing];
```

Comment:
`[refreshControl endRefreshing];` should be moved to the callback when the activities were completely loaded, here you're removing the loading state immediately after the user start refreshing, what happen if the network is slow? this would feel like the app is not working properly